src/main/java/net/education/kyivstar/config/HikariConnection.java

```diff
@@ -0,0 +1,35 @@
+package net.education.kyivstar.config;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import static net.education.kyivstar.config.HikariConfig.getInstance;
+
+public class HikariConnection {
```

Comment:
Combining the functionality of HikariConnection and HikariConfig classes into a single class or renaming them will lead to a cleaner and more cohesive design. Make a **HikariConnectionManager** - responsible for managing the HikariCP connection pool and providing connections.
This consolidation reduces redundancy and simplifies the design by encapsulating all connection management functionality in a single class.

Reply:
+